common/include/pcl/point_cloud.h

```diff
@@ -547,6 +646,16 @@ namespace pcl
         height = 1;
       }
 
+      /** \brief Insert a new point in the cloud, at the end of the container.
+        * \note This assumes the user would correct the details later on!
+        * \param[in] pt the point to insert
+        */
+      inline void
+      transient_push_back (const PointT& pt)
```

Comment:
Most of the `transient_*`-functions could be avoided by adding a new parameter with default value to the existing functions, e.g. `push_back (const PointT& pt, bool no_resize = false)`. I definitely see the need to do push_back/emplace_back/insert/etc without changing width and height, but I think adding many new functions would hurt the clarity.